db/attachment_test.go

```diff
@@ -139,3 +138,39 @@ func TestAttachmentForRejectedDocument(t *testing.T) {
 	assert.True(t, err != nil, "Expect error when attempting to retrieve attachment document after doc is rejected.")
 
 }
+
+func TestAttachmentRetrievalUsingRevCache(t *testing.T) {
+
+	testBucket := base.GetTestBucketOrPanic()
+	defer testBucket.Close()
+	bucket := testBucket.Bucket
+
+	context, err := NewDatabaseContext("db", bucket, false, DatabaseContextOptions{})
+	assert.NoError(t, err, "Couldn't create context for database 'db'")
+	defer context.Close()
+	db, err := CreateDatabase(context)
+	assert.NoError(t, err, "Couldn't create database 'db'")
+
+	// Test creating & updating a document:
+	rev1input := `{"_attachments": {"hello.txt": {"data":"aGVsbG8gd29ybGQ="},
+                                    "bye.txt": {"data":"Z29vZGJ5ZSBjcnVlbCB3b3JsZA=="}}}`
+	_, err = db.Put("doc1", unjson(rev1input))
+	assert.NoError(t, err, "Couldn't create document")
+
+	rev1output := `{"_attachments":{"bye.txt":{"data":"Z29vZGJ5ZSBjcnVlbCB3b3JsZA==","digest":"sha1-l+N7VpXGnoxMm8xfvtWPbz2YvDc=","length":19,"revpos":1},"hello.txt":{"data":"aGVsbG8gd29ybGQ=","digest":"sha1-Kq5sNclPz7QV2+lfQIuc6R7oRu0=","length":11,"revpos":1}},"_id":"doc1","_rev":"1-54f3a105fb903018c160712ffddb74dc"}`
+	gotbody, err := db.GetRev("doc1", "1-54f3a105fb903018c160712ffddb74dc", false, []string{})
+	assert.NoError(t, err, "Couldn't get document")
+	assert.Equal(t, rev1output, tojson(gotbody))
+
+	getCount, countErr := base.GetExpvarAsInt("syncGateway_db", "document_gets")
+	assert.NoError(t, countErr, "Couldn't retrieve document_gets expvar")
+	assert.Equal(t, 0, getCount)
```

Comment:
This works OK in isolation, but when this test runs as part of a suite, expvars are carried over between tests.
Should we consider resetting then via expvarMap.Init() when a test database is created?